src/simd/avx2.rs

```diff
@@ -1,28 +1,27 @@
 use crate::iter::Bytes;
 
-pub enum Scan {
-    /// Returned when an implementation finds a noteworthy token.
-    Found,
-    /// Returned when an implementation couldn't keep running because the input was too short.
-    TooShort,
+#[cfg(target_arch = "x86")]
```

Comment:
What requires defining these stubs? It looks like the method signatures are different... But that would be a compile error, so are these ever even built?

Reply:
The sig mismatch is a mistake, I forgot to update it as I shifted the interface to be `math_x_vectored`. We indeed don't build it in our tests, with neon support it could be helpful to arm64/arm CI possibly x86 whilst we're at it.

The stub is "required" because it's expected by the interface and both x86/x64 paths expect avx2 functions to exist but then be guarded by the feature check. I don't see any reason why we could not just skip avx2 on x86

I'll address that in a follow up if you're onboard